### PR TITLE
Remove callbacks from Geometry and simplify exact representations

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

**Review by BarbourSmith — APPROVED** [review]
This looks nice:

image

This looks maybe wrong:

image

Same with a couple of these actually...

What is it supposed to do?

image

**Comment by pentacular** [comment]
Yes, these are wrong. :)

I put it up as "wip" because I was having trouble seeing what was failing.

Those diffs should be gone now.

**Comment by BarbourSmith** [comment]
Very nice :) 

Looks great to me 👍 👍

**Review by BarbourSmith — APPROVED** [review]

File: algorithm/cgal/Unfold.h [hunk 1]
```diff
@@ -73,15 +79,15 @@ static int Unfold(
                   computeInverseSegmentTransform(s1, s2, Vector(0, 0, 1));
               geometry->segments(edge).emplace_back(s1, s2);
               geometry->setTransform(edge, t.inverse());
-              emit_tag(edge, std::string("unfold:edge"));
+              tags.emplace_back(edge, "unfold:edge");
```
Comment (BarbourSmith) [inline algorithm/cgal/Unfold.h:82]: What is the current status of unfold?

**Review by BarbourSmith — APPROVED** [review]

File: algorithm/cgal/Unfold.h (hunk 1 above)
Comment (pentacular) [reply algorithm/cgal/Unfold.h:82]: Works, but haven't really had much use of it so far.

**Pull request merged**